tests/models/test_openai.py

```diff
@@ -3560,6 +3562,166 @@ async def test_openai_reasoning_in_thinking_tags(allow_model_requests: None):
     )
 
 
+def test_azure_prompt_filter_error(allow_model_requests: None) -> None:
+    body = {
+        'error': {
+            'code': 'content_filter',
+            'message': 'The content was filtered.',
+            'innererror': {
+                'code': 'ResponsibleAIPolicyViolation',
+                'content_filter_result': {'hate': {'filtered': True, 'severity': 'high'}},
+            },
+        }
+    }
+
+    mock_client = MockOpenAI.create_mock(
+        APIStatusError(
+            'content filter',
+            response=httpx.Response(status_code=400, request=httpx.Request('POST', 'https://example.com/v1')),
+            body=body,
+        )
+    )
+
+    m = OpenAIChatModel('gpt-5-mini', provider=AzureProvider(openai_client=cast(AsyncAzureOpenAI, mock_client)))
+    agent = Agent(m)
+
+    try:
+        agent.run_sync('bad prompt')
+    except ContentFilterError as e:
```

Comment:
This test will succeed if no exception is raised :) `with pytest.raises(...) with exc_info` let's you access the `exc_info` after the `with` context.